java/src/main/java/com/genexuscore/genexus/common/SdtGridState_InputValuesItem.java

```diff
@@ -0,0 +1,208 @@
+/*
+               File: genexus.common.SdtGridState_InputValuesItem
+        Description: GridState
+             Author: GeneXus Java Generator version 17_0_0-144773
+       Generated on: September 16, 2020 14:15:57.79
+       Program type: Callable routine
+          Main DBMS: Oracle
+*/
+package com.genexuscore.genexus.common ;
+import com.genexuscore.*;
+import com.genexus.*;
+import com.genexus.xml.*;
+import com.genexus.search.*;
+import com.genexus.webpanels.*;
+import java.util.*;
+
+public final  class SdtGridState_InputValuesItem extends GXXMLSerializable implements Cloneable, java.io.Serializable
+{
+   public SdtGridState_InputValuesItem( )
+   {
+      this(  new ModelContext(SdtGridState_InputValuesItem.class));
+   }
+
+   public SdtGridState_InputValuesItem( ModelContext context )
+   {
+      super( context, "SdtGridState_InputValuesItem");
+   }
+
+   public SdtGridState_InputValuesItem( int remoteHandle ,
+                                        ModelContext context )
+   {
+      super( remoteHandle, context, "SdtGridState_InputValuesItem");
+   }
+
+   public SdtGridState_InputValuesItem( StructSdtGridState_InputValuesItem struct )
+   {
+      this();
+      setStruct(struct);
+   }
+
+   private static java.util.HashMap mapper = new java.util.HashMap();
+   static
+   {
+   }
+
+   public String getJsonMap( String value )
+   {
+      return (String) mapper.get(value);
+   }
+
+   public short readxml( com.genexus.xml.XMLReader oReader ,
+                         String sName )
+   {
+      short GXSoapError = 1 ;
+      formatError = false ;
+      sTagName = oReader.getName() ;
+      if ( oReader.getIsSimple() == 0 )
+      {
+         GXSoapError = oReader.read() ;
+         nOutParmCount = (short)(0) ;
+         while ( ( ( GXutil.strcmp(oReader.getName(), sTagName) != 0 ) || ( oReader.getNodeType() == 1 ) ) && ( GXSoapError > 0 ) )
+         {
+            readOk = (short)(0) ;
+            if ( GXutil.strcmp2( oReader.getLocalName(), "Name") )
+            {
+               gxTv_SdtGridState_InputValuesItem_Name = oReader.getValue() ;
+               if ( GXSoapError > 0 )
+               {
+                  readOk = (short)(1) ;
+               }
+               GXSoapError = oReader.read() ;
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 64:72, 73:81.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.

Reply:
About the he following lines: 64:72, 73:81. That loads the gxTv_SdtGridState_InputValuesItem_Name and then gxTv_SdtGridState_InputValuesItem_Value. And in case the first if success, the oReader.read() moves the cursor to the next node. I don't see it is possible to reduce that code.